File: 1258-59-2375-construct-smallest-number-from-di-string/1258-59-2375-construct-smallest-number-from-di-string.py
class Solution:
    def smallestNumber(self, pattern: str) -> str:
        stk,ans = "",""
        for i in range(0,len(pattern)+1):
            stk+=str(i+1)
            if i == len(pattern) or pattern[i] == 'I':
                    ans+=stk[::-1]
                    stk=""
        return ans
            

File: 1258-59-2375-construct-smallest-number-from-di-string/1258-59-2375-construct-smallest-number-from-di-string.java
class Solution {
    public String smallestNumber(String pattern) {
        int n = pattern.length();
        StringBuilder stk = new StringBuilder(), ans = new StringBuilder();
        for (int i = 0; i <= n; i++) {
            stk.append(i + 1);
            if (i == n || pattern.charAt(i) == 'I') {
                ans.append(stk.reverse());
                stk.setLength(0);
            }
        }
        return ans.toString();
    }
}
